compiler/coreSyn/CoreLint.hs

```diff
@@ -1227,20 +1227,16 @@ lintCoreAlt lookup_scrut scrut_ty scrut_weight alt_ty alt@(DataAlt con, args, rh
         -- We've already check
       lintL (tycon == dataConTyCon con) (mkBadConMsg tycon con)
     ; let { con_payload_ty = piResultTys (dataConRepType con) tycon_arg_tys
-          ; (_, pred, con_args, _) = dataConSig con
-          ; ex_tvs = dataConExTyVars con
-          -- Weights of the things that the data con brings into scope
-          -- MattP: TODO: Move this into DataCon
-          ; weights = (replicate (length ex_tvs + length pred) Omega)
-                        ++ map weightedWeight (con_args) }
+          ; ex_tvs_n = length (dataConExTyVars con)
+          ; weights = replicate ex_tvs_n Omega ++
+                      map weightedWeight (dataConRepArgTys con) }
 
         -- And now bring the new binders into scope
     ; lintBinders CasePatBind args $ \ args' -> do
     {
     rhs_ue <- lintAltExpr rhs alt_ty ;
     let scrut_usage = lookup_scrut rhs_ue
-    -- This goes wrong, see GADT1 test, not sure if weight even matters
-    -- here
+--    ; pprTrace "lintCoreAlt" (ppr weights $$ ppr args' $$ ppr (dataConRepArgTys con) $$ ppr (dataConSig con)) ( return ())
```

Comment:
Dead code?

Reply:
Code that I've typed out multiple time to debug what is going on in this area so I elected to leave it there for the time being in case it is needed again.